File: programmers/js/basic/181830.js
// 정사각형으로 만들기
function solution(arr) {
  const rowLen = arr[0].length;
  const colLen = arr.length;

  if (rowLen > colLen) {
    for (let i = 0; i < rowLen - colLen; i++) {
      arr.push(Array(rowLen).fill(0));
    }

    return arr;
  }

  if (rowLen < colLen) {
    return arr.map((elem) => {
      for (let i = 0; i < colLen - rowLen; i++) {
        elem.push(0);
      }

      return elem;
    });
  }

  return arr;
}

File: programmers/java/basic/181830.java
// 정사각형으로 만들기

class Solution {
    public int[][] solution(int[][] arr) {
        int maxNum = Math.max(arr.length, arr[0].length);
        int[][] result = new int[maxNum][maxNum];
        
        for (int i = 0; i < arr.length; i++) {
            for (int j = 0; j < arr[0].length; j++) {
                result[i][j] = arr[i][j];
            }
        }
        
        return result;
    }
}